applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -112,11 +112,21 @@ static void on_event_app_data(const lwm2m_carrier_event_t *event)
 		}
 	}
 
-	rsp_send("\r\n#XCARRIEREVT: %d,%d,\"%s\"\r\n", event->type, app_data->buffer_len, uri_path);
-
-	memcpy(slm_data_buf, app_data->buffer, app_data->buffer_len);
+	if (app_data->type == LWM2M_CARRIER_APP_DATA_EVENT_DATA_WRITE) {
+		ret = slm_util_htoa(app_data->buffer, app_data->buffer_len,
```

Comment:
You're right, but I think we can accept the risk of not logging the data if it's more than this limit, because realistically we won't ever receive that much data downlink. From the carrier's perspective, the limit is 512 bytes, so 1024 if in hex string format, so I think that limitation is better than having to allocate another big buffer. I have added a comment here to better clarify this in the code.

Reply:
If this is only for logging purposes, you could truncate the data that goes in, with something like:
MIN(sizeof(slm_data_buf) / 2,  app_data->buffer_len)

Alternatively, please fix the comment as slm_data_buf is not smaller than SLM_CARRIER_APP_DATA_BUFFER_LEN, it is just not 2 times  bigger than it, which would be required to hold the ascii formatting.